### PR TITLE
fix: don't die if cached tarball is corrupted

### PR DESCRIPTION
**Pull request by mnzaki**
related to pnpm/pnpm#600

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   },
   "homepage": "https://github.com/pnpm/tarball-fetcher#readme",
   "dependencies": {
+    "@pnpm/logger": "^1.0.0",
```
Comment (zkochan) [inline package.json]: `@pnpm/logger` should be a peer dependency and installed as devDependency for testing.

**Comment by mnzaki** [comment]
Probably unrelated, but one of the tests is failing for me:
```
# fail when integrity check fails two times in a row
# testing in /tmp/52442cdc5119e7b7a7a47f37a08644a9
not ok 6 should be equal
  ---
    operator: equal
    expected: |-
      'sha1-HssnaJydJVE+rbyZFKc/VAi+enY= integrity checksum failed when using sha1: wanted sha1-HssnaJydJVE+rbyZFKc/VAi+enY= but got sha1-ACjKMFA7S6uRFXSDFfH4aT+4B4Y=. (1194 bytes)'
    actual: |-
      'sha1-HssnaJydJVE+rbyZFKc/VAi+enY= integrity checksum failed when using sha1: wanted sha1-HssnaJydJVE+rbyZFKc/VAi+enY= but got sha512-VuFL1iPaIxJK/k3gTxStIkc6+wSiDwlLdnCWNZyapsVLobu/0onvGOZolASZpfBFiDJYrOIGiDzgLIULTW61Vg==
 sha1-ACjKMFA7S6uRFXSDFfH4aT+4B4Y=. (1194 bytes)'
```

something is producing sha512 instead of sha1?

**Comment by zkochan** [comment]
Thanks for your contribution!

That test is broken because of an update of ssri. I pinned it to `5.0.0` for now. I will look into it later

**Comment by zkochan** [comment]
Weird, your test was failing on Node.js 4 because it did not make any requests to the registry. Seems like on Node 4, even corrupt tarballs are unpacked?! Half the files were unpacked successfully! That's not good, I guess.

**Comment by mnzaki** [comment]
That is weird indeed.... maybe we should recheck integrity before trying to unpack local tarballs? And again I'm not sure why execution even gets to this point when the archive is clearly broken (i.e. when is integrity check done?)

**Comment by zkochan** [comment]
Currently, integrity check is done only during download https://github.com/pnpm/tarball-fetcher/blob/master/src/createDownloader.ts#L157

**Pull request merged**